tests/web/render.rs

```diff
@@ -34,6 +35,58 @@ fn container_is_emptied_upon_drop() {
     assert!(container.first_child().is_none());
 }
 
+/// Originally, dodrio would use go through the className property for SVGs.
+/// 
+/// This is problematic because when SVG elements are created, the className is flagged as a read
+/// only property, so setting it causes an exception to be thrown. Here's an example of how this
+/// happens:
+///
+/// let elem = web_sys::window()
+///     .unwrap()
+///     .document()
+///     .unwrap()
+///     .create_element_ns(Some("http://www.w3.org/2000/svg"), "svg")
+///     .unwrap();
+///
+/// elem.set_class_name("does-not-work"); 
+/// 
+/// -----------------------------------------------------------------------------------------------
+///     
+///     wasm-bindgen: imported JS function that was not marked as `catch` threw an error:
+///     setting getter-only property "className"
+/// 
+/// -----------------------------------------------------------------------------------------------
+/// 
+/// Now, dodrio passes the 'class' attribute of all namespaced elements into set_attribute. This
+/// satisfies the restrictions on SVG and keeps the optimized path for non-namespaced elements
+#[wasm_bindgen_test(async)]
+async fn test_svg_set_class() {
+    let container = create_element("div");
+
+
+    let valid_svg = Rc::new(RenderFn(|cx| {
+        ElementBuilder::new(cx.bump, "svg")
+            .namespace(Some("http://www.w3.org/2000/svg"))
+            .attr("class", "works")
+            .finish()
+    }));
+
+    let vdom = Vdom::new(&container, valid_svg.clone());
+    let weak = vdom.weak();
+
+    weak.render().await.unwrap();
+
+    assert_eq!(
+        "works", 
+        container.first_child()
+            .expect("unable to get svg")
+            .dyn_ref::<web_sys::Element>()
+            .expect("svg should be an element")
+            .get_attribute("class")
+            .expect("unable to get 'class' of svg")
+    );
+}
+
```

Comment:
Here's the regression test. I wanted to test the failure case too, but web_sys doesn't mark `set_class_name` as `catch`, so I wasn't able to figure out how to do it.

You can verify that this works by running the following:
```
$ wasm-pack test --firefox --headless -- --features xxx-unstable-internal-use-only test_svg_set_class
```

Then, verify that the test fails when the new change is disabled:

```rust
// src/change_list/mod.rs - comment out !is_namespaced
    pub fn set_attribute(&mut self, name: &str, value: &str, is_namespaced: bool) {
        debug_assert!(self.traversal_is_committed());
        if name == "class" /* && !is_namespaced*/ {
            let class_id = self.ensure_string(value);
            debug!("emit: set_class({:?})", value);
            self.state.emitter.set_class(class_id.into());
        } else {
            let name_id = self.ensure_string(name);
            let value_id = self.ensure_string(value);
            debug!("emit: set_attribute({:?}, {:?})", name, value);
            self.state
                .emitter
                .set_attribute(name_id.into(), value_id.into());
        }
    }
```

Reply:
BTW, I wasn't sure where to put the test, so let me know if you want to move it somewhere else.

Reply:
this is a fine location -- thanks!